java/arcs/core/util/SchedulerUtils.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.util
+
+import kotlinx.coroutines.CoroutineDispatcher
+
+internal val currentDispatcherThreadLocal = ThreadLocal<CoroutineDispatcher?>()
+
+internal fun setCurrentDispatcher(dispatcher: CoroutineDispatcher?) {
+    currentDispatcherThreadLocal.set(dispatcher)
+}
+
+/** The [Scheduler] dispatcher that the current thread is running in, or null. */
+val currentDispatcher get() = currentDispatcherThreadLocal.get()
+
+/** Returns true if the current thread is executing within the given [dispatcher]. */
+fun currentlyRunningInDispatcher(dispatcher: CoroutineDispatcher): Boolean {
+    return currentDispatcher === dispatcher
+}
```

Comment:
This seems too generic as a public function for what we're using it for... and could be misleading to devs down the road. 

I don't want people to accidentally assume that it's a general utility for determining if the current dispatcher is what is passed as an argument.

Instead - I feel like it'd be better to make it Scheduler-specific, and bonus points if it's part of the Scheduler API itself.  The Handles should have access to the scheduler, but even if they dont - I know the StorageProxy does, and the storage proxy could be what does the enforcement (Rather than `checkPreconditions` in BaseHandle)

Reply:
Ok, I'll move it.